pkg/storage/stores/shipper/compactor/retention/util_test.go

```diff
@@ -158,15 +158,13 @@ func (t *testStore) indexTables() []table {
 }
 
 func (t *testStore) HasChunk(c chunk.Chunk) bool {
-	t.t.Helper()
-	var matchers []*labels.Matcher
-	for _, l := range c.Metric {
-		matchers = append(matchers, labels.MustNewMatcher(labels.MatchEqual, l.Name, l.Value))
+	chunks := t.GetChunks(c.UserID, c.From, c.Through, c.Metric)
+
+	chunkIDs := make(map[string]struct{})
+	for _, chk := range chunks {
+		chunkIDs[chk.ExternalKey()] = struct{}{}
 	}
-	chunks, err := t.Store.Get(user.InjectOrgID(context.Background(), c.UserID),
-		c.UserID, c.From, c.Through, matchers...)
-	require.NoError(t.t, err)
-	return len(chunks) == 1 && c.ExternalKey() == chunks[0].ExternalKey()
+	return len(chunkIDs) == 1 && c.ExternalKey() == chunks[0].ExternalKey()
```

Comment:
We could get multiple chunk ids if the chunk overlaps multiple schema configs because the Cortex code dedupes chunk ids within a store, not across the stores. This change dedupes the chunk ids before checking for having single expected chunk id in the response.